src/async_traits.rs

```diff
@@ -371,53 +390,86 @@ where
         let async_conn = Connection(Self::as_async_conn(&conn).0.clone());
         match f(async_conn).await {
             Ok(value) => {
-                conn.run_with_shared_connection(|conn| {
-                    Conn::TransactionManager::commit_transaction(conn).map_err(E::from)
-                })
-                .await?;
-                Ok(value)
+                match conn
+                    .run_with_shared_connection(|conn| {
+                        Conn::TransactionManager::commit_transaction(conn)
+                            .map_err(|err| E::from(RunError::User(err)))
+                    })
+                    .await
+                {
+                    Ok(()) => Ok(value),
+                    // XXX: we should try to roll this back
+                    Err(RunError::User(err)) => Err(err),
```

Comment:
I believe this is automatically done by the `commit_transaction` method in the case where we return an error.

See: https://docs.diesel.rs/master/src/diesel/connection/transaction_manager.rs.html#451

It *looks* l like `get_transaction_state` could return an error and stop us from propagating an error here, but I think that only happens if the TransactionManager has already been marked "broken, do not use" by a different call. So, ignoring that error-return-path, [if we fail to commit, we automatically perform rollback](https://docs.diesel.rs/master/src/diesel/connection/transaction_manager.rs.html#487-508)

TL;DR: I think this comment is unnecessary, we already do try to roll back.

Reply:
Thanks!

Reply:
Removed the comment.